admin/admin_token.go

```diff
@@ -0,0 +1,40 @@
+package admin
+
+import (
+	"time"
+
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/google/uuid"
+	"github.com/vocdoni/blind-csp/types"
+)
+
+// GenerateAdminToken generates a new admin token for an election lasting one month
+func GenerateAdminToken(electionId types.HexBytes) (string, error) {
+	currentTime := time.Now()
+	dateString := currentTime.Format("200601")
+
+	var data types.HexBytes
+	if err := data.FromString(electionId.String() + dateString); err != nil {
+		return "", err
+	}
+
+	return uuid.NewSHA1(uuid.NameSpaceDNS, data).String(), nil
+}
+
+// ValidateAdminToken validates an admin token for an election
+func ValidateAdminToken(electionId types.HexBytes, adminToken string) (bool, error) {
+	generatedToken, err := GenerateAdminToken(electionId)
+	if err != nil {
+		return false, err
+	}
+
+	return adminToken == generatedToken, nil
+}
+
+// VerifySignature verifies a signature
+func VerifySignature(publicKeyBytes types.HexBytes, data types.HexBytes, signature types.HexBytes) (bool, error) {
```

Comment:
We have our own ethereum compatible crypto library with all helpers you need and a code that ensures it is compatible with the signers we use: `import "go.vocdoni.io/dvote/crypto/ethereum"`

I would suggest using this library and also to check against addresses instead of pubKeys, since it is more standard on Ethereum: https://github.com/vocdoni/vocdoni-node/blob/main/crypto/ethereum/ethereum.go#L252

Reply:
oh! That method is not being used, but will refactor to use the dvote/crypto library